src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,44 @@
 import React from 'react';
+import { Status, StatusType } from '../../types/Status';
+import { useDispatch } from 'react-redux';
+import { setQuery, setStatus } from '../../features/filter';
+import { useAppSelector } from '../../app/hooks';
+
+export const TodoFilter = () => {
+  const dispatch = useDispatch();
+  const { status, query } = useAppSelector(state => state.filter);
+
+  const showOptions = Object.values(StatusType).map(stat => (
+    <option key={stat} value={stat}>
+      {stat[0].toUpperCase() + stat.slice(1)}
+    </option>
+  ));
```

Comment:
The `showOptions` constant is re-calculated on each render of the component. It doesn't depend on any props or state, so you could move it outside of the component to avoid unnecessary calculations. For example, you can define it in the module scope, just after imports.